drivers/hid/i2c-hid/i2c-hid-core.c

```diff
@@ -50,7 +50,7 @@
 #define I2C_HID_QUIRK_NO_IRQ_AFTER_RESET	BIT(1)
 #define I2C_HID_QUIRK_NO_RUNTIME_PM		BIT(2)
 #define I2C_HID_QUIRK_DELAY_AFTER_SLEEP		BIT(3)
-#define I2C_HID_QUIRK_BOGUS_IRQ			BIT(4)
```

Comment:
You are not supposed to touch existing quirks. Just add a new one on top.

Reply:
Right, so your quirk becomes:

`#define I2C_HID_QUIRK_FORCE_TRIGGER_FALLING	BIT(5)
`
and then put your device specific quirk before the match using ANY_ID so that it will be checked first, you device specific quirk should look something like this:

```
{ USB_VENDOR_ID_ELAN, I2C_PRODUCT_ID_ELAN_TOUCHPAD,
  I2C_HID_QUIRK_BOGUS_IRQ | I2C_HID_QUIRK_FORCE_TRIGGER_FALLING },
```

And this needs to be above:

```
	{ USB_VENDOR_ID_ELAN, HID_ANY_ID,
		 I2C_HID_QUIRK_BOGUS_IRQ },
```

Otherwise the HID_ANY_ID match will match first and the code looking up the quirks will stop there.

After fixing this please submit a new version of the patch to the mailinglist. Please Cc:
Hans de Goede <hdegoede@redhat.com> when submitting a new version.

Reply:
Alright, I will fix that.